src/SubFolder.php

```diff
@@ -82,21 +68,45 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
                 $newPath = '/';
             }
 
-            if ($newPath[0] !== '/') {
-                if (!$auto) {
-                    throw new BadUriPrefixException('URI prefix does not match completely.');
-                }
-            } else {
+            if ($newPath[0] === '/') {
                 $request = $request->withUri($uri->withPath($newPath));
-                /** @var string $prefix */
-                $this->uriGenerator->setUriPrefix($prefix);
-
-                if ($this->alias !== null) {
-                    $this->aliases->set($this->alias, $prefix . '/');
+                $this->uriGenerator->setUriPrefix($baseUrl);
+                if ($this->baseUrlAlias !== null) {
+                    $this->aliases->set($this->baseUrlAlias, $baseUrl);
                 }
+            } elseif ($this->prefix !== null) {
+                throw new BadUriPrefixException('URI prefix does not match completely.');
             }
         }
 
         return $handler->handle($request);
     }
+
+    public function getBaseUrl(ServerRequestInterface $request): string
+    {
+        $serverParams = $request->getServerParams();
+        $scriptUrl = $serverParams['SCRIPT_FILENAME'];
+        $scriptName = basename($scriptUrl);
+
+        if (isset($serverParams['PHP_SELF']) && basename($serverParams['PHP_SELF']) === $scriptName) {
+            $scriptUrl = $serverParams['PHP_SELF'];
+        } elseif (
+            isset($serverParams['ORIG_SCRIPT_NAME']) &&
+            basename($serverParams['ORIG_SCRIPT_NAME']) === $scriptName
+        ) {
+            $scriptUrl = $serverParams['ORIG_SCRIPT_NAME'];
+        } elseif (
+            isset($serverParams['PHP_SELF']) &&
+            ($pos = strpos($serverParams['PHP_SELF'], '/' . $scriptName)) !== false
```

Comment:
`str_contains` could be used instead since minimal required version of PHP is 8.0.

Reply:
string position is needed.